roofit/roofit/src/RooChebychev.cxx

```diff
@@ -34,17 +38,76 @@ Chebychev polynomial p.d.f. of the first kind
 
 #include "TError.h"
 
-#if defined(__my_func__)
-#undef __my_func__
-#endif
-#if defined(WIN32)
-#define __my_func__ __FUNCTION__
-#else
-#define __my_func__ __func__
-#endif
-
 ClassImp(RooChebychev);
 
+namespace { // anonymous namespace to hide implementation details
+/// use fast FMA if available, fall back to normal arithmetic if not
+static inline double fast_fma(
+	const double x, const double y, const double z) noexcept
+{
+#if defined(FP_FAST_FMA) // check if std::fma has fast hardware implementation
+   return std::fma(x, y, z);
+#else // defined(FP_FAST_FMA)
+   // std::fma might be slow, so use a more pedestrian implementation
+#pragma STDC FP_CONTRACT ON // hint clang that using an FMA is okay here
+   return (x * y) + z;
+#endif // defined(FP_FAST_FMA)
+}
+
+/// Chebychev polynomials of first or second kind
+enum class Kind : int { First = 1, Second = 2 };
+
+/** @brief ChebychevIterator evaluates increasing orders at given x
+ *
+ * @author Manuel Schiller <Manuel.Schiller@glasgow.ac.uk>
+ * @date 2019-03-24
+ */
+template <typename T, Kind KIND>
+class ChebychevIterator {
+private:
+   T m_last = 1;
+   T m_twox = 0;
+   T m_curr = 0;
+
```

Comment:
The ChebychevIterator is a really good idea. Could you change the member names fit into RooFit's naming convention? E.g. `_last`.